### PR TITLE
Use dash in port range definition

### PR DESCRIPTION
**Pull request by mchf**
https://bugzilla.suse.com/show_bug.cgi?id=1084434

### CODE REVIEW

File: src/modules/FtpServer.rb [hunk 1]
```diff
@@ -443,10 +443,9 @@ def WriteFirewallSettings
       return true if !firewalld.running?
 
       if Ops.get(@EDIT_SETTINGS, "PassiveMode") == "YES"
-        port_range = Ops.add(
-          Ops.add(Ops.get(@EDIT_SETTINGS, "PasMinPort"), ":"),
-          Ops.get(@EDIT_SETTINGS, "PasMaxPort")
-        )
+        raise ArgumentError, "Missing min / max ports for passive ftp" if @EDIT_SETTINGS["PasMinPort"].nil? || @EDIT_SETTINGS["PasMaxPort"].nil?
```
Comment (teclator) [inline src/modules/FtpServer.rb]: I'm fine adding the exception, although the line seems to bee tooooo long and also if `nil` is the only option to be checked probably it is enough checking if it is defined with `unless @EDIT_SETTINGS["PasMinPort"] || @EDIT_SETTINGS["PasMaxPort"]` but if not set would be `nil` or could be also an empty string which is also invalid?

Comment (teclator) [reply src/modules/FtpServer.rb]: Also if we are not going to handle the exception, maybe it is preferable to just report an error and return false

Comment (mchf) [reply src/modules/FtpServer.rb]: I'd stay with exception. I've added that obly bcs of my paranoia from old code. In fact everything should be checked - as it is user's input - here.

File: src/modules/FtpServer.rb [hunk 2]
```diff
@@ -443,10 +443,11 @@ def WriteFirewallSettings
       return true if !firewalld.running?
 
       if Ops.get(@EDIT_SETTINGS, "PassiveMode") == "YES"
-        port_range = Ops.add(
-          Ops.add(Ops.get(@EDIT_SETTINGS, "PasMinPort"), ":"),
-          Ops.get(@EDIT_SETTINGS, "PasMaxPort")
-        )
+        if !valid_port_range(@EDIT_SETTINGS["PasMinPort"], @EDIT_SETTINGS["PasMaxPort"])
```
Comment (teclator) [inline src/modules/FtpServer.rb]: The method expect `String` parameters at least based on documentation, so or adapt the documentation or you could use `@EDIT_SETTINGS.fetch("PasMinPort","")` omitting the nil? check

File: src/modules/FtpServer.rb [hunk 3]
```diff
@@ -914,6 +915,21 @@ def self.publish_variable(name, type)
     publish function: :OptionsSummary, type: "string ()"
     publish function: :Summary, type: "string ()"
     publish function: :AutoPackages, type: "map ()"
+
+  private
+
+    # Checker for pasive ftp port range
```
Comment (teclator) [inline src/modules/FtpServer.rb]: Typo `passive`, what about # Return whether the min and max passive ftp port range are both valid or not

File: src/modules/FtpServer.rb [hunk 4]
```diff
@@ -914,6 +915,21 @@ def self.publish_variable(name, type)
...
+    #
+    # @param [String] min_port lower port of the range
```
Comment (teclator) [inline src/modules/FtpServer.rb]: Commented above

File: src/modules/FtpServer.rb (hunk 2 above)
Comment (mchf) [reply src/modules/FtpServer.rb]: ok, you forced me to go through sources ;-)

Everything interesting is done around here https://github.com/yast/yast-ftp-server/blob/master/src/include/ftp-server/wid_functions.rb#L904

```PasMinPort``` and ```PasMaxPort``` are stored as strings but initialized and validated as integers. So, I'll handle that as integers and change my patch accordingly.

Comment (teclator) [reply src/modules/FtpServer.rb]: But that just validate the entries for the widget, have you really tested that the attributes are Integers at the end?

https://github.com/yast/yast-ftp-server/blob/master/src/include/ftp-server/wid_functions.rb#L922

File: src/modules/FtpServer.rb [hunk 5]
```diff
@@ -913,6 +915,20 @@ def self.publish_variable(name, type)
     publish function: :OptionsSummary, type: "string ()"
     publish function: :Summary, type: "string ()"
     publish function: :AutoPackages, type: "map ()"
+
+  private
+
+    # Checker for passive ftp port range
+    #
+    # @param [Integer] min_port lower port of the range
```
Comment (teclator) [inline src/modules/FtpServer.rb]: still missing the [Integer,nil] in that case, but it is really a Integer?

File: src/modules/FtpServer.rb (hunk 2 above)
Comment (mchf) [reply src/modules/FtpServer.rb]: yes, bcs it otherwise will not pass through ```ValidPasMaxPort```

Comment (teclator) [reply src/modules/FtpServer.rb]: Really? 

```ruby
FtpServer.WriteToEditMap(
        "PasMaxPort",
        Builtins.tostring(UI.QueryWidget(Id("PasMaxPort"), :Value))
)
```
Then I'm lost.

**Pull request merged**